src/osd/PG.cc

```diff
@@ -2662,10 +2665,13 @@ void PG::_update_calc_stats()
 	  misplaced += osd_objects;
       } else {
         // If this peer has more objects then it should, ignore them
-        int64_t osd_backfilled = MIN(num_objects, peer_info[p].stats.stats.sum.num_objects);
-        // Include computed backfilled objects on up nodes
-        object_copies += osd_backfilled;
-        backfilled += osd_backfilled;
+        int64_t osd_backfilled = MIN(num_objects,
+				     peer_info[p].stats.stats.sum.num_objects);
+	if (actingset.size() + num_backfill_shards_seen < pool.info.size) {
```

Comment:
It would have been even better if we counted the backfilling OSDs with the greater number of objects, but this is sufficient.

Reply:
Agree.. and now that I think about it it isn't much more work to do it.